hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/tools/federation/RouterAdmin.java

```diff
@@ -462,6 +481,111 @@ public int run(String[] argv) throws Exception {
     return exitCode;
   }
 
+  private boolean addAllMount(String[] parameters, int i) throws IOException {
+    List<AddMountAttributes> addMountAttributesList = new ArrayList<>();
+    while (i < parameters.length) {
+      // Mandatory parameters
+      String mount = parameters[i++];
+      String[] nss = parameters[i++].split(",");
+      String dest = parameters[i++];
+
+      // Optional parameters
+      boolean readOnly = false;
+      boolean faultTolerant = false;
+      String owner = null;
+      String group = null;
+      FsPermission mode = null;
+      DestinationOrder order = DestinationOrder.HASH;
+      while (i < parameters.length && !",".equals(parameters[i])) {
```

Comment:
A lot of this parsing is similar to adding a single entry.
Can we refactor this code so we use the other chunk?

Reply:
I was thinking of the same, the only problem is, here we need to retain the value of `i` and for single add, we don't need to. Let me see how best we can refactor this. On the other hand, would you be fine if I do the refactor as a followup Jira?

Reply:
Done